flepimop/gempyor_pkg/src/gempyor/inference_parameter.py

```diff
@@ -93,150 +157,112 @@ def add_modifier(self, pname, ptype, parameter_config, subpops):
+        """
+        return len(self)
 
-    def get_dim(self):
-        return len(self.pnames)
+    def get_parameters_for_subpop(self, subpop: str) -> list[int]:
+        """
+        Get the indices of parameters relevant for a specific subpopulation.
 
-    def get_parameters_for_subpop(self, subpop: str) -> list:
-        """Returns the index parameters for a given subpopulation"""
-        parameters = []
-        for i, sp in enumerate(self.subpops):
-            if sp == subpop:
-                parameters.append(i)
-        return parameters
+        Args:
+            subpop: The name of the subpopulation to pull parameters indexes for.
 
-    def __len__(self):
-        """
-        so one can use the built-in python len function
+        Returns:
+            A list of indices corresponding to parameters that affect the specified
+            subpopulation.
         """
-        return len(self.pnames)
+        return [i for i, s in enumerate(self.subpops) if s == subpop]
 
-    def draw_initial(self, n_draw=1):
+    def draw_initial(self, n_draw: int = 1) -> npt.NDArray[np.float64]:
         """
         Draws initial parameter values.
 
         Args:
-            n_draw (int): Number of draws, e.g the number of slots or walkers
+            n_draw: Number of draws, e.g the number of slots or walkers.
 
         Returns:
-            np.ndarray: Array of initial parameter values.
+            Array of initial parameter values with shape (`n_draw`, dim).
         """
-        p0 = np.zeros((n_draw, self.get_dim()))
-        for p_idx in range(self.get_dim()):
-            p0[:, p_idx] = self.pdists[p_idx](n_draw)
-
+        dim = len(self)
+        p0 = np.zeros((n_draw, dim))
+        for p_idx in range(dim):
+            p0[:, p_idx] = self.pdists[p_idx].sample(size=n_draw)
         return p0
 
-    # TODO: write a more granular method the return for a single parameter and correct the proposal like we did
-    def check_in_bound(self, proposal) -> bool:
+    def check_in_bound(self, proposal: npt.NDArray[np.float64]) -> bool:
         """
         Checks if the proposal is within parameter bounds.
 
         Args:
             proposal: The proposed parameter values.
 
         Returns:
-            bool: True if the proposal is within bounds, False otherwise.
+            `True` if the proposal is within bounds, `False` otherwise.
         """
-        if self.hit_lbs(proposal=proposal).any() or self.hit_ubs(proposal=proposal).any():
-            return False
-        return True
-
-    def hit_lbs(self, proposal) -> np.ndarray:
-        return np.array((proposal < self.lbs))
-
-    def hit_ubs(self, proposal) -> np.ndarray:
-        """
-        boolean vector of True if the parameter is bigger than the upper bound and False if not
-        """
-        return np.array((proposal > self.ubs))
+        lower_bounds, upper_bounds = zip(*[dist.support for dist in self.pdists])
```

Comment:
how often is check_in_bounds called? i'm not sure about the elimination of ubs, lbs as properties of this object - might still be useful to create and cache those with initialization / parsing, especially if this method is called many times post initialization.

Reply:
Frequently. I'll make the adjustment.